src/Perlang.Interpreter/Typing/TypeCoercer.cs

```diff
@@ -16,14 +16,18 @@ public static class TypeCoercer
             { typeof(Int32), 32 },
             { typeof(Int64), 64 },
 
-            // Double-precision values are 64-bit but can only save 53-bit integers with exact precision
-            { typeof(Double), 32 },
-
```

Comment:
This is the fix. We need to take this one out of this table, to be able to handle integer and floating-point types differently...